internal/pkg/api/handleEnroll.go

```diff
@@ -146,6 +154,30 @@ func (et *EnrollerT) _enroll(
 	}
 
 	agentID := u.String()
+	// only delete existing agent if it never checked in
+	if agent.Id != "" && agent.LastCheckin == "" {
+		zlog.Info().
+			Str("EnrollmentId", enrollmentID).
+			Str("AgentId", agent.Id).
+			Msg("Remove existing agent with the same enrollment_id")
+		// invalidate previous api key
+		err := invalidateAPIKey(ctx, zlog, et.bulker, agent.AccessAPIKeyID)
+		if err != nil {
+			zlog.Debug().
+				Str("EnrollmentId", enrollmentID).
+				Str("AgentId", agent.Id).
+				Str("APIKeyID", agent.AccessAPIKeyID).
+				Msg("Error when trying to invalidate API key of old agent with enrollment id")
+		}
+		// delete existing agent to recreate with new api key
+		err = deleteAgent(ctx, zlog, et.bulker, agent.Id)
```

Comment:
If either of these operations fail I think it would be better to fail the entire request with an InternalServerError?

The agent is able to safely retry the enrollment on failure. If you just proceed here you risk leaving the API key valid or not deleting the agent and ending up with a duplicate.

Either of these operations has a low be realistic chance of failure in the real world. I would just fail the enrollment request and let the agent retry to keep the system in a consistent state.